src/Layers/xrRender/ResourceManager.cpp

```diff
@@ -414,14 +414,20 @@ void CResourceManager::DeferredUpload()
         texture.second->Load();
 #endif
 }
-/*
-void	CResourceManager::DeferredUnload	()
+
+void CResourceManager::DeferredUnload()
 {
-    if (!RDEVICE.b_is_Ready)				return;
-    for (auto t=m_textures.begin(); t!=m_textures.end(); t++)
-        t->second->Unload();
+    if (!RDEVICE.b_is_Ready)
+        return;
+
+#ifndef USE_OGL
+    tbb::parallel_for_each(m_textures, [&](auto m_tex) { m_tex.second->Unload(); });
+#else
+    for (auto& texture : m_textures)
```

Comment:
It probably was just copied from `Load()`:

`Load()` has separate singlethreaded path for OpenGL, because it requires that thread, that loads a texture, must own a context and we don't have efficient mechanism to do that with foreign threads such as TBB threads.

I don't know if `Unload()` can be done in parallel, but it's reasonable to do something to make both `Load()` and `Unload()` parallel.

Reply:
Yeah, the context sharing. A good point to switch to fibers one day